python/ray/autoscaler/_private/autoscaler.py

```diff
@@ -241,20 +249,33 @@ def _update(self):
             return
 
         self.last_update_time = now
-        self.update_worker_list()
+
+        # Query node provider to update workers, all_workers, and all_nodes.
+        self.update_node_lists()
 
         # Remove from LoadMetrics the ips unknown to the NodeProvider.
-        self.load_metrics.prune_active_ips([
-            self.provider.internal_ip(node_id) for node_id in self.all_workers
-        ])
+        self.load_metrics.prune_active_ips(
+            [self.provider.internal_ip(node_id) for node_id in self.all_nodes])
+
+        # Update status strings
+        logger.info(self.info_string())
+        legacy_log_info_string(self, self.workers)
 
         if not self.provider.is_readonly():
             self.terminate_nodes_to_enforce_config_constraints(now)
 
+            if self.disable_node_updaters:
+                self.terminate_unhealthy_nodes(now)
+            else:
+                self.process_completed_updates()
+                self.update_nodes()
+                self.attempt_to_recover_unhealthy_nodes(now)
+                self.set_prometheus_updater_data()
+
         # Dict[NodeType, int], List[ResourceDict]
         to_launch, unfulfilled = (
             self.resource_demand_scheduler.get_nodes_to_launch(
-                self.provider.non_terminated_nodes(tag_filters={}),
+                self.all_nodes,
```

Comment:
why not call this self.non_terminated_nodes?

Reply:
prefer not to re-use the name of the NodeProvider method